mongodbatlas/resource_mongodbatlas_project_api_key.go

```diff
@@ -225,12 +226,23 @@ func resourceMongoDBAtlasProjectAPIKeyRead(ctx context.Context, d *schema.Resour
 
 func resourceMongoDBAtlasProjectAPIKeyUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	conn := meta.(*MongoDBClient).Atlas
+	connV2 := meta.(*MongoDBClient).AtlasV2
+
 	ids := decodeStateID(d.Id())
 	projectID := ids["project_id"]
 	apiKeyID := ids["api_key_id"]
 
-	updateRequest := new(matlas.AssignAPIKey)
+	if d.HasChange("description") {
```

Comment:
nit: Failure in update of description block assign. Maybe description update can be after assignment?